xrpl/models/transactions/amm_bid.py

```diff
@@ -0,0 +1,76 @@
+"""Model for AMMBid transaction type."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+
+from typing_extensions import Final
+
+from xrpl.models.amounts import Amount
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+_MAX_AUTH_ACCOUNTS: Final[int] = 4
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AMMBid(Transaction):
+    """
+    AMMBid is used to place a bid for the auction slot of obtaining trading advantages
+    of an AMM instance.
+
+    An AMM instance auctions off the trading advantages to users (arbitrageurs) at a
+    discounted TradingFee for a 24 hour slot.
+    """
+
+    amm_id: str = REQUIRED  # type: ignore
+    """
+    AMMID is a hash that uniquely identifies the AMM instance.
```

Comment:
nit: it's a little weird/potentially confusing to me that in the docstring it's referred to as `AMMID` and in the actual param it's `amm_id`. A quick scan of other transactions shows that the docstrings usually don't have the name of the param listed - they either just go straight into the explanation (e.g. `A hash that...`), or they use `This field`. IMO that would be a better way of writing these docstrings.

Reply:
Agreed, also xrpl.js goes straight into the explanation so it's a good idea to keep both consistent.